crates/db_schema/src/impls/federation_blocklist.rs

```diff
@@ -8,9 +8,14 @@ use crate::{
 };
 use diesel::{dsl::insert_into, result::Error};
 use diesel_async::{AsyncPgConnection, RunQueryDsl};
+use std::collections::HashMap;
 
 impl FederationBlockList {
-  pub async fn replace(pool: &DbPool, list_opt: Option<Vec<String>>) -> Result<(), Error> {
+  pub async fn replace(
+    pool: &DbPool,
+    list_opt: Option<Vec<String>>,
```

Comment:
This needs to be a single list, not a list and a map.

Reply:
👍  this should happen as a consequence the above